### PR TITLE
imessage-ruby 0.4.0

### PR DESCRIPTION
**Pull request by linjunpop**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Review by SMillerDev — DISMISSED** [review]

**Comment by chenrui333** [comment]
```
  imessage-ruby:
    * Non-executables were installed to "/opt/homebrew/opt/imessage-ruby/bin".
      The offending files are:
        /opt/homebrew/opt/imessage-ruby/bin/apple_scripts
```

**Comment by linjunpop** [comment]
> imessage-ruby:
>     * Non-executables were installed to "/opt/homebrew/opt/imessage-ruby/bin".
>       The offending files are:
>         /opt/homebrew/opt/imessage-ruby/bin/apple_scripts

Hi @chenrui333,

Would you please suggest where should I put these support files? Thanks.

**Comment by SMillerDev** [comment]
Depends on what they are, but unless they are a binary it should not be in `bin`

File: Formula/imessage-ruby.rb [hunk 1]
```diff
@@ -1,22 +1,34 @@
...
-  sha256 "3d93570dd3ff000c51859aaa34c4a8e8b80b86e6bdf3018c1c7f6f8d85939b30"
+  url "https://github.com/linjunpop/imessage/archive/refs/tags/v0.4.0.tar.gz"
+  sha256 "09031e60548f34f05e07faeb0e26b002aeb655488d152dd811021fba8d850162"
   license "MIT"
   head "https://github.com/linjunpop/imessage.git", branch: "master"
 
-  bottle do
-    sha256 cellar: :any_skip_relocation, all: "7c538ac24ec3ce437d868267582bd68aaa500eeb5a9bdbd3d0d80398b4bab19d"
-  end
+  depends_on "ruby" if Hardware::CPU.arm?
```
Comment (SMillerDev) [inline Formula/imessage-ruby.rb:9]: So... it doesn't use ruby from macOS?

Comment (linjunpop) [reply Formula/imessage-ruby.rb]: Hmm.. it should.

**Comment by linjunpop** [comment]
Hi @SMillerDev, would you please review again? Thanks.

File: Formula/imessage-ruby.rb [hunk 2]
```diff
@@ -1,22 +1,30 @@
...
-  sha256 "3d93570dd3ff000c51859aaa34c4a8e8b80b86e6bdf3018c1c7f6f8d85939b30"
+  url "https://github.com/linjunpop/imessage/archive/refs/tags/v0.4.0.tar.gz"
+  sha256 "09031e60548f34f05e07faeb0e26b002aeb655488d152dd811021fba8d850162"
   license "MIT"
   head "https://github.com/linjunpop/imessage.git", branch: "master"
 
-  bottle do
-    sha256 cellar: :any_skip_relocation, all: "7c538ac24ec3ce437d868267582bd68aaa500eeb5a9bdbd3d0d80398b4bab19d"
-  end
-
```
Comment (SMillerDev) [inline Formula/imessage-ruby.rb:12]: Please don't modify the bottle block

File: Formula/imessage-ruby.rb [hunk 3]
```diff
@@ -1,22 +1,30 @@
...
+  uses_from_macos "ruby"
 
   def install
-    system "rake", "standalone:install", "prefix=#{prefix}"
+    ENV["GEM_HOME"] = libexec
+    ENV["GEM_PATH"] = libexec
+
+    system "gem", "build", "imessage.gemspec", "-o", "imessage-#{version}.gem"
+    system "gem", "install", "--local", "--verbose", "imessage-#{version}.gem", "--no-document"
+    system "gem", "info", "imessage"
```
Comment (SMillerDev) [inline Formula/imessage-ruby.rb:17]: Why this one?

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**